gearcmd/worker.go

```diff
@@ -32,31 +33,43 @@ func (conf TaskConfig) Process(job baseworker.Job) ([]byte, error) {
 	// This wraps the actual processing to do some logging
 	log.Println(kayvee.FormatLog("gearcmd", kayvee.Info, "START",
 		map[string]interface{}{"function": conf.FunctionName, "job_id": getJobId(job), "job_data": string(job.Data())}))
-	start := time.Now()
-	err := conf.doProcess(job)
-	end := time.Now()
-	data := map[string]interface{}{
-		"function": conf.FunctionName, "job_id": getJobId(job),
-		"job_data": string(job.Data()), "type": "gauge",
-	}
-	if err != nil {
+
+	for {
+		start := time.Now()
+		err := conf.doProcess(job)
+		end := time.Now()
+		data := map[string]interface{}{
+			"function": conf.FunctionName, "job_id": getJobId(job),
+			"job_data": string(job.Data()), "type": "gauge",
+		}
+		// Return if the job was successful.
+		if err == nil {
+			data["value"] = 1
+			data["success"] = true
+			log.Println(kayvee.FormatLog("gearcmd", kayvee.Info, "END", data))
+			// Hopefully none of our jobs last long enough for a uint64...
+			log.Printf(kayvee.FormatLog("gearcmd", kayvee.Info, "duration",
+				map[string]interface{}{
+					"value": uint64(end.Sub(start).Seconds() * 1000),
+					"type":  "gauge", "function": conf.FunctionName,
+				},
+			))
+			return nil, nil
+		}
 		data["error_message"] = err.Error()
 		data["value"] = 0
 		data["success"] = false
-		log.Println(kayvee.FormatLog("gearcmd", kayvee.Error, "END", data))
-	} else {
-		data["value"] = 1
-		data["success"] = true
-		log.Println(kayvee.FormatLog("gearcmd", kayvee.Info, "END", data))
-		// Hopefully none of our jobs last long enough for a uint64...
-		log.Printf(kayvee.FormatLog("gearcmd", kayvee.Info, "duration",
-			map[string]interface{}{
-				"value": uint64(end.Sub(start).Seconds() * 1000),
-				"type":  "gauge", "function": conf.FunctionName,
-			},
-		))
+		// Return if the job has no more retries.
+		if conf.RetryCount <= 0 {
+			log.Println(kayvee.FormatLog("gearcmd", kayvee.Error, "END", data))
+			return nil, err
```

Comment:
It actually looks like it didn't complain, but I can still change this if you'd like.  

Or `make test` could be using a different linter?